platforms/jvm/plugins-java-base/src/main/java/org/gradle/api/plugins/JavaBasePlugin.java

```diff
@@ -445,7 +445,7 @@ public Directory call() throws Exception {
                 // In the Kotlin plugin, the subclass manages its own field which will be used here.
                 // This was to support tasks that extended AbstractCompile and had their own getDestinationDir().
                 // We actually need to keep this as compile.getDestinationDir to maintain compatibility.
-                legacyValue = compile.getDestinationDir();
+                legacyValue = compile.getDestinationDirectory().getAsFile().get();
```

Comment:
Looking at the Java doc for this class, is this backwards compatibility/fallback doing what it's intended to anymore?

Do we even need this now?

Reply:
No, this shouldn't be needed any more.  I missed removing it.  Removed now.

Reply:
Looks like you didn't push up your changes

Reply:
Oh, push was rejected and I didn't notice.  Fixed.